docs/development.md

```diff
@@ -382,8 +392,180 @@ The `lang` property of the `<script>` block is used to specify the used programm
+Requirments are specified as an array of JavaScript urls. These libraries will then be
+imported using `importScripts`.
+
+For example, to specify the latest [plotly.js](https://plot.ly/javascript/) library you can
+use
+```json
+"requirements": ["https://cdn.plot.ly/plotly-latest.min.js"]
+```
+
+For window plugins, you can also specify **CSS urls**, these need to end with `.css`.
+
+For example, to use the [W3.CSS framework](https://www.w3schools.com/w3css/), you can specify
+```json
+"requirements": ["https://www.w3schools.com/w3css/4/w3.css"]
+```
+
+ImJoy hosts **commonly used and tested libraries** in a dedicated [GitHub repository](https://github.com/oeway/static.imjoy.io).
+You can refer to all files contained in the `docs` folder with a simple url: `https://static.imjoy.io` + `RelativePathInDocs`.
+
+For example, the file `FileSaver.js` in the folder `static.imjoy.io/docs/js/`
+can be referenced as
+```json
+"requirements": ["https://static.imjoy.io/js/FileSaver.js"]
+```
+
+### Native Python plugins
+
+Requirements are defined as a list of strings.
+A prefix is used to specify the supported requirement types: `conda:`, `pip:` and `repo:`.
+
+The general syntax is `"requirements": ["prefix:requirementToInstall"]`. The table below
+lists all supported requirements, the actual command being executed by ImJoy and
+an example.
+
+Prefix   | Command            | Example
+---------|--------------------|----------------------------------------
+`conda` | `conda install -y` | `"conda:scipy==1.0"`
+`pip`   | `pip install`      | `"pip:scipy==1.0"`
+`repo`  | `git clone` (new repo)  <br> `git pull` (existing repo)|  `"repo:https://github.com/userName/myRepo"`
+`cmd`  | Any other command  |  `"cmd:pip install -r myRepo/requirements.txt"`
+
+
+Some **important considerations**:
+
+*  If **no prefix is used**, requirements will be treated as `pip` libraries.
+   ```json
+   "requirements": ["numpy", "scipy==1.0"]
+   ```
+
+*  If a **virtual environment** is defined by setting `env`, all `pip` and `conda`
+   packages will be installed to this environment. For more information see the
+   dedicated section on [virtual environments](development?id=virtual-environments).
+
+*  You can list multiple requirements either directly as one string after a prefix:
+   ```json
+   "requirements": ["conda:numpy scipy==1.0"]
+   ```
+   or as separate strings in one list:
+
+   ```json
+   "requirements": ["conda:numpy", "conda:scipy==1.0"]
+   ```
+*  Different requirement types can be combined into one list.
+
+   ```json
+    "requirements": ["conda:numpy", "pip:scipy==1.0", "repo:https://github.com/userName/myRepo"]
+    ```
+
+*  `pip` also supports installation directly from a git url, assuming that the
+   repository contains `setup.py`. In this case, we recommend
+   specifying the [GitHub release](https://help.github.com/articles/creating-releases/)
+   of your library, e.g. with `#egg=myRepo`. This ensures that the correct
```

Comment:
This part about #egg seems wrong, because the format for pip is :
`pip install git+https://github.com/{ username }/{ reponame }@{ tag name }#egg={ desired egg name }`

So the tag should be specified with `@XXXX`,  and `#egg=XXX` is just for specify the pip egg name, typically the python module name.

Reference:
https://pip.pypa.io/en/latest/reference/pip_install/#git